From fa71737489c4455f8bfdcb583311e975800e0895 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:00:52 -0400
Subject: [fs] btrfs: add missing error check in incremental send

Message-id: <1407179109-13714-166-git-send-email-sandeen@redhat.com>
Patchwork-id: 87093
O-Subject: [RHEL7.1 PATCH 165/422] Btrfs: add missing error check in incremental send
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Filipe David Borba Manana <fdmanana@gmail.com>

commit d86477b303da51832002eec1cdec2938c42fccc3
Author: Filipe David Borba Manana <fdmanana@gmail.com>
Date:   Thu Jan 30 13:27:12 2014 +0000

    Btrfs: add missing error check in incremental send

    Function wait_for_parent_move() returns negative value if an error
    happened, 0 if we don't need to wait for the parent's move, and
    1 if the wait is needed.
    Before this change an error return value was being treated like the
    return value 1, which was not correct.

    Signed-off-by: Filipe David Borba Manana <fdmanana@gmail.com>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/send.c b/fs/btrfs/send.c
index 9dde971..70272e1 100644
--- a/fs/btrfs/send.c
+++ b/fs/btrfs/send.c
@@ -3227,7 +3227,10 @@ verbose_printk("btrfs: process_recorded_refs %llu\n", sctx->cur_ino);
      * dirs, we always have one new and one deleted
      * ref. The deleted ref is ignored later.
      */
-    if (wait_for_parent_move(sctx, cur)) {
+    ret = wait_for_parent_move(sctx, cur);
+    if (ret < 0)
+     goto out;
+    if (ret) {
      ret = add_pending_dir_move(sctx,
            cur->dir);
      *pending_move = 1;
-- 
1.7.1